lib/Optimizer/Lower.cpp

```diff
@@ -94,13 +94,14 @@ void lowerRegressionGradNode(Function *F, RegressionGradNode &node) {
 }
 
 void lowerFullyConnectedNode(Function *F, FullyConnectedNode &FC) {
-  auto xDim = flattenCdr(FC.getInput().getType()->dims());
-  auto wDim = FC.getWeights().dims();
+  auto xDim = flattenCdr(FC.getInput().getType()->dims(), FC.getAxis());
```

Comment:
If you end up leaving Axis and AxisW, then you'll also need to support them in `lowerFullyConnectedGradNode()`